simpledotorg/simple-server
# Consolidate hypertension reports

**Story card:** [sc-8281](https://app.shortcut.com/simpledotorg/story/8281/consolidate-hypertension-reports)

## Because

Diabetes reports are going to be launched in a new "Diabetes" tab on the reports page. This makes the information architecture of the reports a little weird, with several HTN tabs and one DM tab.

## This addresses

This PR consolidates the Overview, Details, and Cohort Reports pages into a single "Hypertension" page in the reports section. Here are some additional notes about the change.

* The restructuring is only effective when the `diabetes_management_reports` feature flag is turned on. Several controller actions and views and stuff can be deleted when we are removing the feature flag.
* We have chosen to simply display both monthly and quarterly cohort reports one after the other, instead of using a dropdown to toggle between them
* There is some substantial overlap of controller code between the different sections of the hypertension report. We will schedule a cleanup for later. We want to make the restructure right away so we can get ahead of the launch of the DM reports.

## Test instructions

* Turn off the DM flag, all tabs of the reports tab should look the same
* Turn on the DM flag, there should now be only "Hypertension" and "Diabetes" tabs on the reports page (also "Progress" for facility reports)
* The "Hypertension" report should contain the overview stuff ( 4 cards, treatment outcome bar graph, etc), details stuff (registrations and followups, LTFU, etc), and monthly and quarterly cohort reports.

## Screenshots
<img width="1414" alt="Screen Shot 2022-05-12 at 10 40 47 PM" src="https://user-images.githubusercontent.com/4241399/168145981-c1f0ccdd-5fca-44e2-8c84-3d6ffd0f0672.png">
<img width="1399" alt="Screen Shot 2022-05-12 at 10 40 57 PM" src="https://user-images.githubusercontent.com/4241399/168145986-92ecb2be-cd6e-4d9e-9cdf-7cc9d2573209.png">
<img width="1281" alt="Screen Shot 2022-05-12 at 10 41 06 PM" src="https://user-images.githubusercontent.com/4241399/168145995-c1c29ebf-8d0b-4dc6-bbf4-5ba664d5149f.png">
